File: Dante1/Module1.04/Zadanie4.24/main.c
#include <stdio.h>

int main()
{
    int vector1[100];
    int size1 = 0;

    int vector2[100];
    int size2 = 0;

    printf("Enter: ");

    for (int i = 0; i < 100; i++)
    {
        int check = scanf("%i", &vector1[i]);
        if (check != 1)
        {
            printf("Incorrect input");
            return 1;
        }
        if (vector1[i] == 0)
        {
            break;
        }
        size1++;
    }

    if (size1 < 1)
    {
        printf("\nEnter: ");
        for (int i = 0; i < 100; i++)
        {
            int check = scanf("%i", &vector2[i]);
            if (check != 1)
            {
                printf("Incorrect input");
                return 1;
            }
            if (vector2[i] == 0)
            {
                break;
            }
            size2++;
        }
        printf("Not enough data available");
        return 2;
    }

    while (getchar() != '\n');

    printf("\nEnter: ");
    for (int i = 0; i < 100; i++)
    {
        int check = scanf("%i", &vector2[i]);
        if (check != 1)
        {
            printf("Incorrect input");
            return 1;
        }
        if (vector2[i] == 0)
        {
            break;
        }
        size2++;
    }

    if (size2 < 1)
    {
        printf("Not enough data available");
        return 2;
    }

    printf("\n");

    int temp = 0;
    for(int i = 0; i < size1; i++)
    {
        for(int j = 0; j < size2; j++)
        {
            if (vector1[i] == vector2[j])
            {
                int temp_shit = 0;
                for (int k = 0; k < i; k++)
                {
                    if (vector1[k] == vector1[i])
                    {
                        temp_shit = 1;
                        break;
                    }
                }

                if (!temp_shit)
                {
                    printf("%i ", vector1[i]);
                    temp = 1;
                }
                break;
            }
        }
    }
    if(temp == 0)
    {
        printf("Nothing to show");
    }

    return 0;
}

File: Dante1/Module1.04/Zadanie4.24/main.rs
fn main()
{
    let mut array_one:[i32; 100] = [0; 100];
    let mut size_of_array_one: usize = 0;
    
    println!("Enter: ");
    
    let mut input: String = String::new();
    
    for _i in 0 .. 101
    {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Incorrect input");
        let number: i32 = input.trim().parse().expect("Incorrect number");
        
        if number == 0
        {
            break;
        }
        
        array_one[size_of_array_one] = number;
        size_of_array_one += 1;
    }
    
    let mut array_two:[i32; 100] = [0; 100];
    let mut size_of_array_two: usize = 0;
    
    if size_of_array_one < 1
    {
        println!("Enter: ");
        
        for _i in 0 .. 101
        {
            input.clear();
            std::io::stdin().read_line(&mut input).expect("Incorrect input");
            let number: i32 = input.trim().parse().expect("Incorrect number");
            
            if number == 0
            {
                break;
            }
            
            array_two[size_of_array_two] = number;
            size_of_array_two += 1;
        }
        
        panic!("Not enough data available");
    }
    
    println!("Enter: ");
    for _i in 0 .. 101
    {
        input.clear();
        std::io::stdin().read_line(&mut input).expect("Incorrect input");
        let number: i32 = input.trim().parse().expect("Incorrect number");

        if number == 0
        {
            break;
        }

        array_two[size_of_array_two] = number;
        size_of_array_two += 1;
    }
    
    if size_of_array_two < 1
    {
        panic!("Not enough data available");
    }
    
    let mut temp: u8 = 0;
    
    for _i in 0 .. size_of_array_one
    {
        for _j in 0 .. size_of_array_two
        {
            if array_one[_i] == array_two[_j]
            {
                let mut helper: u8 = 0;
                
                for _k in 0 .. _i
                {
                    if array_one[_k] == array_one[_i]
                    {
                        helper = 1;
                        break;
                    }
                }
                
                if helper == 0
                {
                    println!("{} ", array_one[_i]);
                    temp = 1;
                }
                
                break;
            }
        }
    }
    
    if temp == 0
    {
        println!("Nothing to show");
    }
    return;
}
